### DISCUSSION TITLE
Magisk reporting install not done after flashing OTA

### DISCUSSION THREAD

**Discussion by MvRoo** [chenxiaolong/avbroot#353] [Q&A] [answered]
I was following your guide to create a magisk patched OTA (GrapheneOS). I was able to successfully boot into both a rootless OTA and a Magisk patched OTA. After installing the Magisk app (v37) in the flashed Magisk OTA and starting it, Magisk shows the following error: "your device needs additional setup for Magisk to work properly. Do you want to proceed and reboot?". With the unlocked bootloader, when I proceed and reboot, Magisk works properly and I am able to successfully run the 'adb shell su -c 'dmesg | grep libfs_avb'' command.

Two questions: 
- Is this expected behavior, or could it be that I did something wrong in the steps irt the magisk-preinit-device?
- When I lock the bootloader, can I have magisk run the fix and reboot my device, or will I soft/hardbrick my device?

Appreciate your help and kudos on avbroot and the clear instructions!

**Comment by pascallj** [comment, marked as answer]
> * Is this expected behavior, or could it be that I did something wrong in the steps irt the magisk-preinit-device?

I can't remember exactly if this is expected behaviour, but it is at least safe to continue.

> * When I lock the bootloader, can I have magisk run the fix and reboot my device, or will I soft/hardbrick my device?

As long as you keep the checkbox "Allow OEM unlocking" in developer settings enabled, there is a very small risk in hardbricking your device. You can always unlock your bootloader again and start over. You will lose your data (but you will with locking your bootloader as well), but the device will be functional.

**Reply by MvRoo** [reply]
Thanks for confirming!

**Reply by chenxiaolong** [reply]
Yep, the `your device needs additional setup for Magisk to work properly` message is perfectly normal. It's updating the executables in `/data/adb/` with the latest versions (or initial versions for the initial install). You'll see this message every time you flash a patched OTA that includes a Magisk update.

As far as Magisk goes, the main thing to avoid is anything that mentions `direct install`.
